### DISCUSSION TITLE
How do I do Sonic 3&K Act and Zone transitions?

### DISCUSSION THREAD

**Discussion by Fritzmeinzer** [Techokami/SonicWorldsNext#167] [Q&A]
What Im trying to do is something like the two videos shown. Do I just have an Act Container and instance Act 2 after Act 1? Also how can I have it so the Level (including tilemap, objects etc) are placed directly after the last tile of the Act 1?

https://github.com/Techokami/SonicWorldsNext/assets/70732371/2076e3a4-42d9-4396-b32b-cae6f7a41053

https://github.com/Techokami/SonicWorldsNext/assets/70732371/0f65c7e7-67aa-45cc-b533-7aff22467394

**Comment by Raptorq1** [comment]
> What Im trying to do is something like the two videos shown. Do I just have an Act Container and instance Act 2 after Act 1? Also how can I have it so the Level (including tilemap, objects etc) are placed directly after the last tile of the Act 1?
> 
>  Angel.Island.Act.1.to.2.mp4 
>  AIZ.to.Hydrocity.mp4

I'm trying to do the same

**Comment by DimensionWarped** [comment]
For the basic concept, I'd do this...

1. Make your level just be the full tile map containing both areas. (Or hell, three if you want to outstay an act's welcome like Metropolis Zone) -- Sonic 3 does things this way anyway... which you can see for yourself using the debug mode in certain situations.
2. Have a node for each act of your map -- this node should contain all of your active objects for the given section of the level. They are effectively your active object folders for each act.
3. Create a new type of sign-post that indicates the level stage is advancing instead of actually transitioning which scene you go to when the end of stage score tabulation finishes. This will also kick off the same UI you'd get when loading a new stage. You may need to tweak some of the code in that HUD to make you be able to pop up a new number... though it's probably a small change if anything.
4. When you enter act 1, pause all non-act-1 per-act nodes. This should disable all of the objects for the other acts. I'm not sure yet whether this means they won't render or if they just won't process or what. There may be more complexity to this process than I'm letting on since I haven't played with any of the stuff in this area yet.
5. When transitioning to the new act, disable all the per-act nodes and enable the node for the new act you are going into.

Obviously when you switch levels you'll need to set up barriers and the like to prevent the player from returning to sections of the previous level, but most of the time that's as simple as setting a new camera boundary.

Honestly I think it's pretty straight forward in principle... though I can't promise it won't be without frustration.

**Reply by Raptorq1** [reply]
You may need to tweak some of the code in that HUD to make you be able to pop up a new number... though it's probably a small change if anything.

What would the code be
